app/bundles/LeadBundle/Entity/LeadFieldRepository.php

```diff
@@ -279,28 +279,36 @@ public function compareValue($lead, $field, $value, $operatorExpr, ?string $fiel
                     // multiselect field values are separated by `|` and must be queried using regexp
                     $operator = str_starts_with($operatorExpr, 'not') ? 'NOT REGEXP' : 'REGEXP';
 
-                    $expr = $q->expr()->andX(
+                    $expr = $q->expr()->and(
                         $q->expr()->eq('l.id', ':lead')
                     );
 
                     // require all multiselect values in condition
-                    $andExpr = $q->expr()->andX();
+                    $andExpr = null;
                     foreach ($value as $v) {
                         $v = $q->expr()->literal(
                             InputHelper::clean($v)
                         );
 
                         $v = trim($v, "'");
-                        $andExpr->add(
+
+                        if (null === $andExpr) {
+                            $andExpr = $q->expr()->and(
+                                $property." $operator '\\\\|?$v\\\\|?'"
+                            );
+                            continue;
+                        }
+
+                        $andExpr->with(
```

Comment:
Is there a difference from doing this like
```suggestion

                        $expr->with(
```
?

Reply:
@escopecz using `with` require an existing statement, either `and` or `or`, so with can not be used directly, in sum it is not allowed to create an empty statement

Reply:
@escopecz what about this solution 

```
                    // require all multiselect values in condition
                    $andExpr = [];
                    foreach ($value as $v) {
                        $v = $q->expr()->literal(
                            InputHelper::clean($v)
                        );

                        $v = trim($v, "'");
                        $andExpr[] = $property." $operator '\\\\|?$v\\\\|?'";
                    }

                    $expr = $expr->with($q->expr()->and(...$andExpr));

                    $q->where($expr)
                        ->setParameter('lead', (int) $lead);
```

Reply:
Solved in this commit 813aaeecdfad20dea9f4240bd48eff425362fda2